webapp/src/components/viewHeader/viewHeader.tsx

```diff
@@ -51,6 +51,15 @@ const ViewHeader = React.memo((props: Props) => {
 
     const hasFilter = activeView.filter && activeView.filter.filters?.length > 0
 
+    const onCloseFilter = () => {
+        const myCall = (event: Event) => {
+            if (showFilter && !buttonRef.current?.contains(event.target as Node)) {
+                setShowFilter(false)
+            }
+        }
+        document.addEventListener('click', myCall)
+        return () => document.removeEventListener('click', myCall)
```

Comment:
I don't get how this un-register the event listener. This is returning the function that unregister the event listener, but this is not an effect, so... who is calling the returned function?